src/PointwiseFunctions/AnalyticData/Xcts/BinaryWithGravitationalWaves.hpp

```diff
@@ -0,0 +1,628 @@
+            detail::Tags::IntegralTerm<DataType>,
+            detail::Tags::PostNewtonianConjugateMomentum3<DataType>,
+            detail::Tags::PostNewtonianExtrinsicCurvature<DataType>,
+            ::Tags::deriv<
+                Xcts::Tags::ShiftBackground<DataType, 3, Frame::Inertial>,
+                tmpl::size_t<3>, Frame::Inertial>,
+            gr::Tags::Conformal<gr::Tags::EnergyDensity<DataType>, 0>,
+            gr::Tags::Conformal<gr::Tags::StressTrace<DataType>, 0>,
+            gr::Tags::Conformal<gr::Tags::MomentumDensity<DataType, 3>, 0>,
+            // For initial guesses
+            Xcts::Tags::ConformalFactorMinusOne<DataType>,
+            Xcts::Tags::LapseTimesConformalFactorMinusOne<DataType>,
+            Xcts::Tags::ShiftExcess<DataType, 3, Frame::Inertial>>,
+        hydro_tags<DataType>>>;
+
+template <typename DataType>
+struct BinaryWithGravitationalWavesVariables
+    : CommonVariables<DataType,
+                      BinaryWithGravitationalWavesVariablesCache<DataType>> {
+  static constexpr size_t Dim = 3;
+  using Cache = BinaryWithGravitationalWavesVariablesCache<DataType>;
+  using Base =
+      CommonVariables<DataType,
+                      BinaryWithGravitationalWavesVariablesCache<DataType>>;
+  using Base::operator();
+
+  BinaryWithGravitationalWavesVariables(
+      std::optional<std::reference_wrapper<const Mesh<Dim>>> local_mesh,
+      std::optional<std::reference_wrapper<const InverseJacobian<
+          DataType, Dim, Frame::ElementLogical, Frame::Inertial>>>
+          local_inv_jacobian,
+      const tnsr::I<DataType, 3>& local_x, const double local_mass_left,
+      const double local_mass_right, const double local_xcoord_left,
+      const double local_xcoord_right, const double local_ymomentum_left,
+      const double local_ymomentum_right,
+      const double local_attenuation_parameter)
+      : Base(std::move(local_mesh), std::move(local_inv_jacobian)),
+        x(local_x),
+        mass_left(local_mass_left),
+        mass_right(local_mass_right),
+        xcoord_left(local_xcoord_left),
+        xcoord_right(local_xcoord_right),
+        ymomentum_left(local_ymomentum_left),
+        ymomentum_right(local_ymomentum_right),
+        attenuation_parameter(local_attenuation_parameter) {
+    momentum_left.get(0) = 0.;
+    momentum_left.get(1) = ymomentum_left;
+    momentum_left.get(2) = 0.;
+    momentum_right.get(0) = 0.;
+    momentum_right.get(1) = ymomentum_right;
+    momentum_right.get(2) = 0.;
+  }
+
+  const tnsr::I<DataType, 3>& x;
+  const double mass_left;
+  const double mass_right;
+  const double xcoord_left;
+  const double xcoord_right;
+  const double ymomentum_left;
+  const double ymomentum_right;
+  const double attenuation_parameter;
+  const double separation = xcoord_right - xcoord_left;
+  const std::array<double, 3> normal_lr{{-1., 0., 0.}};
+  tnsr::I<DataType, 3> momentum_left = x;
+  tnsr::I<DataType, 3> momentum_right = x;
```

Comment:
It is like that so that I can use the dot_product() in my expressions. Using  double it doesn't compile.

Reply:
Ok I suggest you use `tnsr::I<double, 3>` and do the dot product manually then, because otherwise you're creating a tensor over the entire grid filled with a constant number. That seems a bit too wasteful :)